.github/workflows/book.yml

```diff
@@ -0,0 +1,30 @@
+name: mdbook
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build-and-upload-to-s3:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@master
+
+      - name: Setup mdBook
+        uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: '0.3.5'
+
+      - run: mdbook build
+        working-directory: book
+
+      - uses: jakejarvis/s3-sync-action@master
+        with:
+          args: --follow-symlinks --delete
+        env:
+          AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
```

Comment:
Oh perhaps we could set this to `AWS_S3_BOOK_BUCKET` in case we decide to make an action that publishes the cargo docs too?

Reply:
You may not have seen it, but we also have this:

https://lighthouse-docs.sigmaprime.io/

It's probably well out of date by now.

Reply:
I renamed the bucket name secret.

As for https://lighthouse-docs.sigmaprime.io/, I think it's redundant.  The go to place for code documentation is either crates.io or docs.rs.  I think we should publish there by the Principle of Least Astonishment ;)  Creating a GitHub `v*.*.*` tag seems like a proper trigger for that.

Reply:
We unfortunately can't publish on crates.io because we have references to local crates (e.g., `some_crate = { path = "../some_crate" }`). We _could_ go and publish everything as a separate crate but that becomes very painful when you're making inter-crate changes (e.g., need to republish or modify the `Cargo.toml` to test).

I always assumed that you can only publish on docs.rs if you publish on crates.io.. Perhaps this is a bad assumption.

Reply:
> Creating a GitHub `v*.*.* tag` seems like a proper trigger for that.

What's this tag? I'm not familiar with it.

Reply:
> I always assumed that you can only publish on docs.rs if you publish on crates.io.. Perhaps this is a bad assumption.

That's a valid assumption despite how my wording might have indicated otherwise ;)

> What's this tag? I'm not familiar with it.

Oh that's just a shell-inspired lingo for tags like `v0.1.0`, `v0.1.1`, etc.  GitHub Actions support such wildcards: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

> We unfortunately can't publish on crates.io because we have references to local crates (e.g., some_crate = { path = "../some_crate" }). We could go and publish everything as a separate crate but that becomes very painful when you're making inter-crate changes (e.g., need to republish or modify the Cargo.toml to test).

Didn't realize that.  In that case, a link to https://lighthouse-docs.sigmaprime.io/ on the crates.io page should do.  I've created an issue for publishing code docs: https://github.com/sigp/lighthouse/issues/897.  Feel free to prioritize it.

Reply:
> Didn't realize that. In that case, a link to https://lighthouse-docs.sigmaprime.io/ on the crates.io page should do.

I might be misunderstanding, but we can't even have a crates.io page for Lighthouse. `cargo publish` will always fail due to those path-based crate links :(

Reply:
Ah, I see now 👍